### ISSUE TITLE
review file load errors from monthly update

### ISSUE THREAD

**Issue by dougkdev** [FreeUKGen/FreeCENMigration#256]
After freecen1 gets updated (monthly) and we load the new/modified freecen1 files into freecen2, there are typically a handful of files that won't load into freecen2 because of errors in the files not caught by freecen1 (freecen2 has more error-checking than freecen1). I periodically take a look at the error files and let Brenda know what the issues are so they can be resolved for the next freecen1 build.  

Vino needs to know how to do this, so I would like Vino to go through the process of finding and resolving the issues this month (with my guidance) so she knows what to look for and becomes familiar with the freecen1 files that get loaded into the system, and the types of problems that frequently occur.

E5, U6, C5S10 P:26

**Comment by richpomfret** [comment]
@dougkdev could we start adding in sizes to tasks?

**Comment by richpomfret** [comment]
A number of load errors found by Brenda (for Lemon to assist with).

**Comment by richpomfret** [comment]
Problem with Aberdeen - ongoing.

**Comment by benwbrum** [comment]
Aberdeen is cleared up, but there are still more problems.  @FreecenBren and @Vino-S will be looking at this.

**Comment by benwbrum** [comment]
No time last sprint; postponing for tonight's update to review results from that.

**Comment by richpomfret** [comment]
Update has been completed. @FreecenBren now looking at this.

**Comment by richpomfret** [comment]
Remains in progress - still potential issues.

**Comment by Captainkirkdawson** [comment]
An on line action has been added for system administrators . FreeCen Errors
https://www.freecen.org.uk/freecen_errors?
This shows both record errors as well as file upload errors.
@Vino-S @FreecenBren It would seem that the majority of the file load errors arise from a reversal of the year and piece number
![errors.JPG](https://images.zenhubusercontent.com/5b68ae684b5806bc2bdefcf0/e68ec14d-f851-41ad-81b4-26813526f1b9)

**Comment by FreecenBren** [comment]
Not at home at the moment so cannot check too much.

What I can say is that no where do any transcribers type in the piece
number or Census year.

When a spreadsheet was been transcribed using the columns as and when there
is data to enter from the census image, the spreadsheet is saved according
to the census year and piece number.

For example if a piece number is called 1234 and the census year is 1861.
Then the piece will be saved as RG091234.
All Census years from 1861 are given an RG number as follows
1841 and 1851 use HO107 as that is how the British Census started until
they (NA) decided in 1861 to use RG instead which stands for Registration
Districts.

1861 starts as RG09xxxx
1871 starts as RG10xxxx
1881 starts as RG11xxxx
1891 starts as RG12xxxx

1851 starts as HO51xxxx
1841 starts as HO107xxx
xxxx being used is the PIECE Number.
All of these are saved this way according to the National Archives details.
We therefore always refer to these as the PIECE NUMBERS.
Nothing will get upload if saved in any other way.

Freecen was devised to work on these numbers.
Each VLD are all numbered the same with the VLD extension.

Example. 1891 for LONDON would be uploaded as RG120001.VLD.

No way is any PIECE number duplicated due to the unique numbering using the
Census year devised by the NA.
When I send any new VLD.s to Lemon I have to send them in folder with the
CHP name used.
So I create a folder say London, LND and add all the  NEW VLD into the LND
folder. I create a folder for each County that I have VLD,s from each COORD.
I have to do the same for each VLD we have amended, so Lemon knows to
replace with the new one.
Lemon struggles otherwise if they are not sent that way and makes his job
harder. So I assume that Lemon uploads them by the CHP cod

So looking at the list you have attached, that the system somehow has
Mixed the FILE and YEAR  columns up as no way can it come from the VLD as
neither of those of are  in the VLD  itself, as both the Census year and
pIECE are part of the name of the VLD.

The only time we get a whole VLD on the Error list is if we send Lemon a
new VLD and we had already got one uploaded for that name.
It then fails as a duplicate. That is why we make sure that Lemon has all
the amended VLD’s in a different  folder when I add them to DROPBOX.

NB: The reason why the 1851 census is not named the same as 1841 is because
1851 use 4 Digits and not 3 like the 1841 and the FC system cannot  accept
the HO107xxxx so we changed it to name them HO51xxxx

Off out for the evening so will look more tomorrow at you table as below.
Brenda

**Comment by Captainkirkdawson** [comment]
@FreecenBren I think the question we need answered is a) do the unloaded files in the above list amount to the missing 59.902 records and 2) why were they not processed. Certainly the message provided is inadequate and needs fixing. Also we need to include the file name in that report to help you

**Comment by Captainkirkdawson** [comment]
@FreecenBren just wanted to emphasize that you should log into FC ie FC2 and use the FreeCen Errors action to view the error report and I will be making enhancements to that to clarify the report 
![errors.JPG](https://images.zenhubusercontent.com/5b68ae684b5806bc2bdefcf0/75fcaaae-2404-490b-b6fb-95f24eec3601)

**Comment by Captainkirkdawson** [comment]
**An analysis of the log for the August 12 update show that**

0) The update only set out to process **2**  pieces
----DELETED VLDS (count: 0)--
[]
----MULTIPLE VLDS (count: 0)--
[]
----NEW VLDS (count: 2)--
[{"chapman"=>"BRK", "file"=>"/raid/freecen2/freecen1/pieces/BRK/rg121002.vld", "base"=>"rg121002.vld", "base_up"=>"RG121002.VLD", "digest"=>"4ff6b7c6eba5a17793c25bcaad25af6b"}, {"chapman"=>"GLS", "file"=>"/raid/freecen2/freecen1/pieces/GLS/rg091797.vld", "base"=>"rg091797.vld", "base_up"=>"RG091797.VLD", "digest"=>"8bddb40c98f9d6d2767d5355832670e9"}]
----MODIFIED VLDS (count: 0)--
[]
----UNCHANGED VLDS (count: 9734)--

1) The BKS piece 1002 crashed the system because of the entry string `Cook € Domestic Servant` containing an in valid UTF-8. @FreecenBren is fixing the data stream.  A story  #725 has been written to resolve this issue more elegantly. 
2) The GLS piece rg091797.vld crashed the system because of `undefined method `match' for nil:NilClass  at freecen1_vld_transformer.rb:23 in block in transform_file_record`. **@FreecenBren you cannot fix. So do not investigate. ** A story #726 has been written to resolve this crash.

3) ** There is an earlier log for the August 12 update with the same time stamp that appears to be damaged and clearly shows processing of more VLDs at least 34 @Vino-S can you shed any light on these 2 logs for what appears to be 1 update and why the earlier one is incomplete**

4) **There is no further comment in the log about the other 18 error files so I went back to the log for the July 15 update. This is quite interesting.**

5) The update set out to process **28**  pieces
----DELETED VLDS (count: 0)--
[]
----MULTIPLE VLDS (count: 0)--
[]
----NEW VLDS (count: 28)--
[{"chapman"=>"BRK", "file"=>"/raid/freecen2/freecen1/pieces/BRK/rg121002.vld", "base"=>"rg121002.vld", "base_up"=>"RG121002.VLD", "digest"=>"4ff6b7c6eba5a17793c25bcaad25af6b"}, {"chapman"=>"ERY", "file"=>"/raid/freecen2/freecen1/pieces/ERY/rg093557.vld", "base"=>"rg093557.vld", "base_up"=>"RG093557.VLD", "digest"=>"95eb67f8d549db25d66479b60b18775f"}, {"chapman"=>"GLS", "file"=>"/raid/freecen2/freecen1/pieces/GLS/rg091797.vld", "base"=>"rg091797.vld", "base_up"=>"RG091797.VLD", "digest"=>"8bddb40c98f9d6d2767d5355832670e9"}, {"chapman"=>"HAM", "file"=>"/raid/freecen2/freecen1/pieces/HAM/RG120865.VLD", "base"=>"RG120865.VLD", "base_up"=>"RG120865.VLD", "digest"=>"4ebd1ae8bea5f760f8f8169512c92749"}, {"chapman"=>"LAN", "file"=>"/raid/freecen2/freecen1/pieces/LAN/rg092663.vld", "base"=>"rg092663.vld", "base_up"=>"RG092663.VLD", "digest"=>"2aec2888628e361d6dd0a75572e1f60e"}, {"chapman"=>"LAN", "file"=>"/raid/freecen2/freecen1/pieces/LAN/rg092682.vld", "base"=>"rg092682.vld", "base_up"=>"RG092682.VLD", "digest"=>"a3d13c0aa031e1edc2a93a84a119ff8d"}, {"chapman"=>"LAN", "file"=>"/raid/freecen2/freecen1/pieces/LAN/rg092696.vld", "base"=>"rg092696.vld", "base_up"=>"RG092696.VLD", "digest"=>"e15380a7f2f4cc73fc063d4a19ec7897"}, {"chapman"=>"LAN", "file"=>"/raid/freecen2/freecen1/pieces/LAN/rg092705.vld", "base"=>"rg092705.vld", "base_up"=>"RG092705.VLD", "digest"=>"0b115a48068886ff2bb95198d01f3c51"}, {"chapman"=>"LAN", "file"=>"/raid/freecen2/freecen1/pieces/LAN/rg092720.vld", "base"=>"rg092720.vld", "base_up"=>"RG092720.VLD", "digest"=>"5e51f4ef97eeb2a8d57079a87be6d142"}, {"chapman"=>"LAN", "file"=>"/raid/freecen2/freecen1/pieces/LAN/rg092721.vld", "base"=>"rg092721.vld", "base_up"=>"RG092721.VLD", "digest"=>"b81d003f0fd083aaa6a542a06aed30a8"}, {"chapman"=>"LAN", "file"=>"/raid/freecen2/freecen1/pieces/LAN/rg092722.vld", "base"=>"rg092722.vld", "base_up"=>"RG092722.VLD", "digest"=>"95b6dacca7e25d70f9bcdd6c7fe8c5bd"}, {"chapman"=>"LAN", "file"=>"/raid/freecen2/freecen1/pieces/LAN/rg092728.vld", "base"=>"rg092728.vld", "base_up"=>"RG092728.VLD", "digest"=>"b3815b94fb7e25af867b261b4ef1e3f7"}, {"chapman"=>"MDX", "file"=>"/raid/freecen2/freecen1/pieces/MDX/RG090798.VLD", "base"=>"RG090798.VLD", "base_up"=>"RG090798.VLD", "digest"=>"9858722f4de196a1a960f81589f0cc52"}, {"chapman"=>"NTT", "file"=>"/raid/freecen2/freecen1/pieces/NTT/rg103530.vld", "base"=>"rg103530.vld", "base_up"=>"RG103530.VLD", "digest"=>"287c6ca48a979f3eaf5cab781833534d"}, {"chapman"=>"NTT", "file"=>"/raid/freecen2/freecen1/pieces/NTT/rg103534.vld", "base"=>"rg103534.vld", "base_up"=>"RG103534.VLD", "digest"=>"b6110868b1afdea8dd8e28d1660e5046"}, {"chapman"=>"SFK", "file"=>"/raid/freecen2/freecen1/pieces/SFK/RG121477.VLD", "base"=>"RG121477.VLD", "base_up"=>"RG121477.VLD", "digest"=>"17ae9d9c85b45ac912cff56433b6cfba"}, {"chapman"=>"SOM", "file"=>"/raid/freecen2/freecen1/pieces/SOM/RG121886.VLD", "base"=>"RG121886.VLD", "base_up"=>"RG121886.VLD", "digest"=>"1135784c5eb3d9e696ca5bf883e85bfe"}, {"chapman"=>"SOM", "file"=>"/raid/freecen2/freecen1/pieces/SOM/RG121889.VLD", "base"=>"RG121889.VLD", "base_up"=>"RG121889.VLD", "digest"=>"2d13027b7a21cd443c92dbea16bb46cf"}, {"chapman"=>"SOM", "file"=>"/raid/freecen2/freecen1/pieces/SOM/RG121890.VLD", "base"=>"RG121890.VLD", "base_up"=>"RG121890.VLD", "digest"=>"db1e4f7699ed77d1250aeb0e2dfeb146"}, {"chapman"=>"WRY", "file"=>"/raid/freecen2/freecen1/pieces/WRY/rg093302.vld", "base"=>"rg093302.vld", "base_up"=>"RG093302.VLD", "digest"=>"123f51fbfe346f9bd85a10a892def065"}, {"chapman"=>"WRY", "file"=>"/raid/freecen2/freecen1/pieces/WRY/rg093303.vld", "base"=>"rg093303.vld", "base_up"=>"RG093303.VLD", "digest"=>"53319a17b5f85b74a8e6ec53fd5fcbd0"}, {"chapman"=>"WRY", "file"=>"/raid/freecen2/freecen1/pieces/WRY/RG104320.VLD", "base"=>"RG104320.VLD", "base_up"=>"RG104320.VLD", "digest"=>"3d2149e9f629543daed838e6076eeeb1"}, {"chapman"=>"WRY", "file"=>"/raid/freecen2/freecen1/pieces/WRY/RG104534.VLD", "base"=>"RG104534.VLD", "base_up"=>"RG104534.VLD", "digest"=>"fdcb582e9bbfac1a82996d32e8742ca4"}, {"chapman"=>"WRY", "file"=>"/raid/freecen2/freecen1/pieces/WRY/RG104576.VLD", "base"=>"RG104576.VLD", "base_up"=>"RG104576.VLD", "digest"=>"0952a9964deba0e1f187046b2fc4f52c"}, {"chapman"=>"WRY", "file"=>"/raid/freecen2/freecen1/pieces/WRY/RG104603.VLD", "base"=>"RG104603.VLD", "base_up"=>"RG104603.VLD", "digest"=>"75c808841b560edd3d8af7dfa0d9d25e"}, {"chapman"=>"WRY", "file"=>"/raid/freecen2/freecen1/pieces/WRY/RG104671.VLD", "base"=>"RG104671.VLD", "base_up"=>"RG104671.VLD", "digest"=>"e2ea80679abb27c9bc75aaca27a17a20"}, {"chapman"=>"WRY", "file"=>"/raid/freecen2/freecen1/pieces/WRY/RG104719.VLD", "base"=>"RG104719.VLD", "base_up"=>"RG104719.VLD", "digest"=>"afc6ad5f94599d74a01cb0aa201c9ad7"}, {"chapman"=>"WRY", "file"=>"/raid/freecen2/freecen1/pieces/WRY/RG104725.VLD", "base"=>"RG104725.VLD", "base_up"=>"RG104725.VLD", "digest"=>"e93214dcba15918c91aa136c5d1540b7"}]
----MODIFIED VLDS (count: 0)--
[]
----UNCHANGED VLDS (count: 9672)--
 (not listing unchanged VLDs)

6) There is an identical crash to 1) above `The BKS piece 1002 crashed the system because of the entry string Cook € Domestic Servant containing an in valid UTF-8.`

7) ERY/rg093557.vld, GLS/rg091797.vld, HAM/RG120865.VLD,  LAN/rg092663.vld, LAN/rg092682.vld, LAN/rg092696.vld,

**Comment by FreecenBren** [comment]
Interesting findings about what FC2 upload issues.

Lemon had to take down the amended VLDs first anyway, before he can replace
them with the amendments.  There were 13 VLDs amended last month for 5
different Counties of LAN, CAM, PER, WAR and WRY.

Also 36 new VLDs (from 13 diff counties) + the 13 amended VLDs  be uploaded
again, a total of 49 VLDS to be uploaded.
How many of those did  FC2 upload?
So interesting why FC2 only had 28?

This is the link that shows all the VLDs  that were uploaded into FC1.
It does not show the 13 Amendments though.
https://freecen1.freecen.org.uk/recent.html

Also this one that FC1 produces  https://freecen1.freecen.org.uk/errors.html
that Kirk has now fixed for us on FC2.

The last FC2 upload was a bit later due to Vino being ill.

We do not get a list from FC2 that shows what gets uploaded after each
upload. Does one exist?
It would be a nice to have as it would save me having to do a comparison
one against the other which takes time as well as having to wait for the
FC2 upload. Just a thought but open to suggestions.

**Comment by Captainkirkdawson** [comment]
@FreecenBren I did complete that analysis after you commented so may wish to read to the end

**Comment by Captainkirkdawson** [comment]
@FreecenBren commented b email
`They are all still in the Amended Dropbox folder that Lemon picks them up
from so this will have to be referred to Lemon.
The 8 LAN ones have been a problem since January and beyond.
We could not find the problem
So 3 weeks ago I stripped one of them right back myself and found the
problem. So then did all 8 and we kept our fingers crossed.
They all went through as you see.

The others were done by myself and COORDS according to the County.
I will check the LIN ones tomorrow (off out now) and see which version is
in FreeCEN ONE ie. Before the amendment or after the amendment.
Then we will know if Lemon did them all or not.`
